File: pages/base_page.py
from playwright.sync_api import Page
import allure

class BasePage:
    """Base page object with common functionality."""

    def __init__(self, page: Page):
        self.page = page

    @allure.step("Navigate to {url}")
    def navigate_to(self, url: str):
        """Navigate to a URL."""
        self.page.goto(url)

    @allure.step("Wait for page load")
    def wait_for_load(self):
        """Wait for page to load completely."""
        self.page.wait_for_load_state("networkidle")

    @allure.step("Take screenshot")
    def take_screenshot(self, name: str):
        """Take a screenshot."""
        self.page.screenshot(path=f"screenshots/{name}.png")

File: pages/base_page.js
const { test } = require('@playwright/test');

class BasePage {
  constructor(page) {
    this.page = page;
  }

  async navigateTo(url) {
    await test.step(`Navigate to ${url}`, async () => {
      await this.page.goto(url);
    });
  }

  async waitForLoad() {
    await test.step('Wait for page load', async () => {
      await this.page.waitForLoadState('networkidle');
    });
  }

  async takeScreenshot(name) {
    await test.step('Take screenshot', async () => {
      await this.page.screenshot({ path: `screenshots/${name}.png` });
    });
  }
}

module.exports = { BasePage };
